### PR TITLE
sessionctx/variable: Remove "advanced" memory sysvars

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/32286

Problem Summary:

The "advanced" memory configuration variables are deprecated, and unused, but are not removed. This can lead to confusion for users as it did in https://github.com/pingcap/tidb/issues/32286

Since we now have a way of removing sysvars, *and* this change is only intended for master (with the next release being a major version) it seems appropriate that they are removed.

### What is changed and how it works?

Unused system variables are removed, it is functionality neutral otherwise.
There is some small cleanup because the names of unused variables are no longer exported, which revealed a couple of uses in the server (non-issues) that are also cleaned up.

### Check List

Tests 

- [x] Unit test
Covered by existing tests for removed variables behavior. Existing tests pass showing variable removal is safe.
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility
Very minor incompatibility, since these currently do nothing, and the removed variable mechanism works pretty well.

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
The unused system variables `tidb_mem_quota_hashjoin`, `tidb_mem_quota_mergejoin`, `tidb_mem_quota_sort`, `tidb_mem_quota_topn`, `tidb_mem_quota_indexlookupreader`, `tidb_mem_quota_indexlookupjoin` have been removed. Please use `tidb_mem_quota_query` to configure memory usage instead.
```

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -1553,9 +1543,6 @@ func (s *SessionVars) GetSystemVar(name string) (string, bool) {
 	} else if name == ErrorCount {
 		return strconv.Itoa(int(s.SysErrorCount)), true
 	}
-	if name == TiDBSlowLogMasking {
-		name = TiDBRedactLog
-	}
```
Comment (morgo) [inline sessionctx/variable/session.go:1558]: This variable was previously removed, so this code path is not reached. It was detected when the `TiDBSlowLogMasking` was no longer exported.

File: sessionctx/variable/removed.go [hunk 2]
```diff
@@ -21,11 +21,30 @@ package variable
...
+	tiDBMemQuotaHashJoin           = "tidb_mem_quota_hashjoin"
+	tiDBMemQuotaMergeJoin          = "tidb_mem_quota_mergejoin"
+	tiDBMemQuotaSort               = "tidb_mem_quota_sort"
+	tiDBMemQuotaTopn               = "tidb_mem_quota_topn"
+	tiDBMemQuotaIndexLookupReader  = "tidb_mem_quota_indexlookupreader"
+	tiDBMemQuotaIndexLookupJoin    = "tidb_mem_quota_indexlookupjoin"
+	tiDBEnableGlobalTemporaryTable = "tidb_enable_global_temporary_table"
+	tiDBSlowLogMasking             = "tidb_slow_log_masking"
+	placementChecks                = "placement_checks"
+)
```
Comment (morgo) [inline sessionctx/variable/removed.go:35]: These were moved in here from `sessionctx/variable/tidb_vars.go`. The comments around them said they were deprecated, but they've been removed. So isolating them to here and no longer exporting seems to be a better fit to truly ensure this is the case.

**Review by XuHuaiyu — APPROVED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**